db/migrate/20190806193607_fill_in_providers.rb

```diff
@@ -0,0 +1,14 @@
+class FillInProviders < ActiveRecord::Migration[5.2]
+  def change
+    User.where('provider IS NULL').find_each do |user|
+      identity = Identity.find_by(email: user.email) if user.email
+      if identity
+        user.update_attribute(:provider, 'identity')
+      elsif user.uid.present?
```

Comment:
I wonder if we ever set `uid` since we never set `provider`.  I'm guessing not but having this `elsif` block is probably good to have even if it might never trigger.

Reply:
But if that's the case then do you think we should add `uid` just like how `provider` is being set on users now?

Reply:
in the case of Identity, uid would be email. This _might_ protect against someone who changes their email getting disconnected from the Identity object, but I think 1) that's not supported and 2) that's pretty out of scope since fixing the dev only Identity provider isn't really the point here.

I check uid not for the Identity case, but for other providers.  SAML or Shibboleth for example would need UID set. We don't want to guess local on those. TBH almost all providers would need to set UID other than the Identity case, making this a nice safety feature for SSO having implementations.

Reply:
I don't think we have ever set `uid` even for SSO omniauth providers.  At least we don't here at IU with a CAS provider and if I'm reading our user model code we only set username and email (and now provider with this PR).

Reply:
I think you're making the case that UID should not be changed then and the code is ok as is? or am I misunderstanding something?

Reply:
~I guess I'm suggesting that we make `VisibleProviders.first[:provider]` be the `else` clause instead of `local`.  Since users don't have `uid`s then line 8 will never be executed and I think the configured provider is what users will expect more than `local`. Hmmm...but there could be incorrect entries then if users started with LTI...so maybe `local` makes the most sense.~  So coming full circle, I'm fine with how this is written now.